interview-questions.md

```diff
@@ -7,44 +7,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-  Your answer:
+  Your answer: If you forget to add the foreign key, you can fix the issue through a migration. The foreign key would be named cohort_id and would reside in the Student model.
 
-  Researched answer:
+  Researched answer: A foreign key is the integer corresponding to the primary key in the main model and is what allows for a referential relationship between both tables. The primary key is automatically created but the foreign key must be added to the non-primary table.
 
 
 
 2. Which RESTful routes must always be passed params? Why?
 
-  Your answer:
+  Your answer: The RESTful routes which must always be passed params are show, update and delete. These need params in order to effect an action on a particular item.
 
-  Researched answer:
+  Researched answer: Show lists one item in a particular model and as such needs to know exactly which item it is showing. Update modifies a particular item and delete destroys a particular item so each likewise need that specific item id in order to effect the action.
 
 
 
 3. Name three rails generator commands. What is created by each?
 
-  Your answer:
+  Your answer: Three rails generator commands are generate model, generate migration, and generate rspec. Generate model creates a database model which can be used to store information, generate migration allows you to modify an existing database, and generate rspec allows you to run tests on your rails code.
 
-  Researched answer:
+  Researched answer: Generate model can be used to create ActiveRecord models. When creating the model you can also generate some fields for it. Generate migration allows you to alter your database schema over time, essentially creating a new version of an existing database. Generate rspec installs the rspec gem, which is a testing framework for a rails application.
```

Comment:
Looking for the Generate resource command but generate rspec is very creative well done!